src/shared/parameter_v2/BUILD

```diff
@@ -0,0 +1,17 @@
+load(":parameter.bzl", "generate_dynamic_parameters")
+
+generate_dynamic_parameters(
```

Comment:
I opted to give this rule conditional behavior based on the file extension it's generating, so that we can easily make 2 targets (one for CPP and one for C). Are people fine with this or would they rather have two separate rules entirely? (In either case I expect the Python stuff to stay the same).

Reply:
Seems like you're proposing that only dynamic parameters cpp header files get the .hpp extension and c header files get the .h extension. I don't like this idea, because the rest of our codebase has .h for cpp header files, so I think it would be too easy to people to intend a cpp header file and forget the .hpp. In effect, we'd have a special exception for dynamic params. I would prefer the rules set up in such a way that we explicitly declare which type of param file we want (separate rules or as an explicit argument to the rule).

Reply:
Opted to split into separate rules (used macros to avoid duplicating most of the implementation).